### PR TITLE
Added metric for days until certificate expiry

### PR DESCRIPTION
**Pull request by ninataneja**
Added a metric for the number of days a certificate is valid for (until expiry), with labels for the certificate serial number and the corresponding cluster URL. This metric will allow us to create an alert for when the certificate expiry date is within a certain # of days (for example, send an alert if a certificate is expiring in the next 15 days)

### CODE REVIEW

File: pkg/controller/certificaterequest/renew_certificate.go [hunk 1]
```diff
@@ -59,6 +60,9 @@ func (r *ReconcileCertificateRequest) ShouldRenewOrReIssue(reqLogger logr.Logger
 		timeDiff := notAfter.Sub(currentTime)
 		daysCertificateValidFor := int(timeDiff.Hours() / 24)
 		shouldRenew := daysCertificateValidFor <= renewBeforeDays
+		certSerialNum := certificate.SerialNumber
+
+		localmetrics.UpdateCertExpiryDateMetric(cr.Spec.WebConsoleURL, certSerialNum, daysCertificateValidFor)
```
Comment (tparikh) [inline pkg/controller/certificaterequest/renew_certificate.go]: The dimensions on this metric don't make much sense. You should log following dimensions for each metrics

- cr.Name
- cr.Spec.ACMEDNSDomain
- cert_expiry_month
- cert_expiry_day
- cert_expiry_year
- cert_expiring_in_days

Webconsole URL and Serial number are not needed.

Comment (fahlmant) [reply pkg/controller/certificaterequest/renew_certificate.go]: I disagree. Labels should only be used to categorize metrics, not have data in them.

Comment (tparikh) [reply pkg/controller/certificaterequest/renew_certificate.go]: The dimensions used here currently don't help in any way with the metrics.

Comment (fahlmant) [reply pkg/controller/certificaterequest/renew_certificate.go]: They categorize the metric by signifying which cert/cluster is expiring.

Comment (tparikh) [reply pkg/controller/certificaterequest/renew_certificate.go]: @fahlmant Yes and the dimensions added in the initial code do not allow for that. We do not want console url. We identify cluster using ACMEBaseDomain.

Comment (fahlmant) [reply pkg/controller/certificaterequest/renew_certificate.go]: Okay, by why cert expiry year, day, month and expiring in days? That will create a new metric for every day

Comment (tparikh) [reply pkg/controller/certificaterequest/renew_certificate.go]: OK I have updated the dimension.

Comment (fahlmant) [reply pkg/controller/certificaterequest/renew_certificate.go]: LGTM :+1:

**Comment by ninataneja** [comment]
found potential bug/issue with Prometheus gauge labels, working on figuring it out

**Comment by c-e-brumm** [comment]
Way too old, will review if needed.

**Pull request closed without merging**